pkg/scheduler/scheduler_test.go

```diff
@@ -44,6 +50,14 @@ func TestCreateScheduler(t *testing.T) {
 			},
 			enableSchedulerEstimator: false,
 		},
+		{
+			name: "scheduler with outOfTree Plugins",
+			opts: []Option{
+				WithOutOfTreeRegistry(karmadaruntime.Registry{
+					"Foo": NewFoo,
```

Comment:
Unit tests don't seem to test the **opt** if  the Foo plugin is registered

Reply:
Yes, hard to verify the result with unit test, integration test would be better. I used to check the right path here, but it seems meaningless, removed.